doc/readDaq.md

```diff
@@ -0,0 +1,151 @@
+# readDaq VI
+
+This VI will read data from analog sensors and status of relays, using NI modules.
+This data will be put into a first-in first-out (FIFO) data structure.
+The name of this structure will be `daqFIFO`.
+
+## DAQ FIFO
+
+The `daqFiFO` is a data structure with an U16 data type defined, 9 elements to store, and configured as a `target-to-host DMA` FIFO, that means the elements will be written in the FPGA and read by the real-time processor.
+
+It is declared as a typedef enum like this:
+
+```c
+{
+   NiFpga_mainFPGA_TargetToHostFifoU16_daqFIFO = 0,
+} NiFpga_mainFPGA_TargetToHostFifoU16;
+```
+## How to get elements from FIFO
+
+The C API provides two functions to get elements from FIFO: (1) [NiFpga_ReadFifoU16()](https://www.ni.com/docs/en-US/bundle/fpga-interface-c-api-ref/page/capi/functions_fifo_read.html) and (2) [NiFpga_AcquireFifoReadElementsU16()](https://www.ni.com/docs/en-US/bundle/fpga-interface-c-api-ref/page/capi/functions_fifo_read_acquire.html).
+
+### 1) NiFpga_ReadFifoU16() function
+
+Here is an example to get elements from the `daqFIFO` using C/C++
+
+```c 
+uint16_t data[9] = {0};
+size_t numberOfElements = 0;
+uint32_t timeout = 0;
+size_t elementsRemaining = 0;
+
+NiFpga_ReadFifoU16(session, NiFpga_mainFPGA_TargetToHostFifoU16_daqFIFO, &data[0], numberOfElements, timeout, &elementsRemaining);
+```
+
+where:
+- `session` handles a currently FPGA session
+- `NiFpga_mainFPGA_TargetToHostFifoU16_daqFIFO` is the FIFO from which to read
+- `data` outputs the data that was read as an array of elements.
+The user will allocate the memory by himself/herself in this method.
+- `numberOfElements` is the number of elements to read
+- `timeout` specifies the time in milliseconds the function waits for available data in the FIFO, if the FIFO is empty
+- `elementsRemaining` outputs the number of elements remaining in the host memory part of FIFO
+
+Note:
+
+- Session has to be oppened with:
+
+```c
+NiFpga_Session session;
+```
+
+- Session must be closed when no longer needed, with:
+
+```c
+NiFpga_Close(session, 0);
+```
+
+### 2) NiFpga_AcquireFifoReadElementsU16() function
+
+Here is an example to get elements from the `daqFIFO` using C/C++
```

Comment:
Here is an example to acquire the elements from `daqFIFO`:

Reply:
Done.